### ISSUE TITLE
test_reboot.py triggers warm reboot failure: BRCM SAI warm recovery failure

### ISSUE THREAD

**Issue by yxieca** [Azure/sonic-buildimage#5885]
**Description**

**Steps to reproduce the issue:**
1. Running nightly test, particularly test_reboot.py
2.
3.

**Describe the results you received:**
warm reboot failed, orchagent crashed.

**Describe the results you expected:**
test succeeded.

**Additional information you deem important (e.g. issue happens only occasionally):**

    **Output of `show version`:**

    ```
master.480-385dfc49
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
Nov 10 14:57:10.064932 str2-dx010-acs-6 INFO swss#restore_neighbor: Error: [Errno 2] No such file or directory: '/sys/class/net/Ethernet64/carrier'
Nov 10 14:57:10.064932 str2-dx010-acs-6 INFO swss#restore_neighbor: Error: [Errno 2] No such file or directory: '/sys/class/net/Ethernet68/carrier'
Nov 10 14:57:11.092573 str2-dx010-acs-6 INFO syncd#syncd: [0] SAI_API_SWITCH:brcm_sai_create_switch:829 SAI Timing: SDK init time 8 seconds
Nov 10 14:57:11.093660 str2-dx010-acs-6 INFO syncd#syncd: [0] SAI_API_TAM:_brcm_sai_eapp_config_init:770 There is no Embedded apps configuration.
Nov 10 14:57:11.105146 str2-dx010-acs-6 INFO syncd#syncd: [0] SAI_API_SWITCH:_brcm_sai_split_horizon_asic_fn_init:153 Split horizon table setting is not needed
Nov 10 14:57:11.112162 str2-dx010-acs-6 INFO syncd#syncd: [0] SAI_API_BUFFER:driverMMUInit:479 Set CPU Tx queue to cosq 7
Nov 10 14:57:11.112253 str2-dx010-acs-6 ERR syncd#syncd: [0] SAI_API_BUFFER:_brcm_sai_setup_cosq_ep_to_ip_map:401 Egress pool value get failed with error Invalid parameter (0xfffffffc).
Nov 10 14:57:11.112303 str2-dx010-acs-6 ERR syncd#syncd: [0] SAI_API_BUFFER:_brcm_sai_map_pool:436 Error -5
Nov 10 14:57:11.112357 str2-dx010-acs-6 ERR syncd#syncd: [0] SAI_API_BUFFER:driverMMUInit:699 Error -5
Nov 10 14:57:11.112407 str2-dx010-acs-6 CRIT syncd#syncd: [0] SAI_API_SWITCH:brcm_sai_create_switch:890 Device MMU Init Failed failed with error Table empty (0xfffffffb).
Nov 10 14:57:11.112454 str2-dx010-acs-6 NOTICE syncd#syncd: :- performWarmRestartSingleSwitch: Warm boot: create switch VID: oid:0x21000000000000 took 8.280075 sec
Nov 10 14:57:11.112504 str2-dx010-acs-6 ERR syncd#syncd: :- performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_FAILURE for VID oid:0x21000000000000
Nov 10 14:57:11.112552 str2-dx010-acs-6 NOTICE syncd#syncd: :- onSyncdStart: on syncd start took 8.290276 sec
Nov 10 14:57:11.112601 str2-dx010-acs-6 ERR syncd#syncd: :- run: Runtime error during syncd init: :- performWarmRestartSingleSwitch: failed to create switch RID: SAI_STATUS_FAILURE for VID oid:0x21000000000000    ```

**Comment by gechiang** [comment]
Additional info about this issue when it failed it was running master.480 image

**Comment by gechiang** [comment]
This is a BRCM SAI issue.
A simpler trigger for this is by running "warm-reboot -vvv" which can also reproduce this issue.
This seems to be BRCM ASIC dependent.
It fails on DUT that has ASIC BCM56960_B1  (Tomahawk)
but works fine with ASIC BCM56971_B0 (Tomahawk2)
CASE CS00011465498 filed with BRCM support.
On Tomahawk it seems SAI is unable to retrieve "bcmCosqControlUCEgressPool" value as part of its validation before performing QOS Egress pool to Ingress Pool mapping operation.

**Comment by gechiang** [comment]
Fix is available as part of the 4.2.1.5 BRCM SAI code patched drop...
https://github.com/Azure/sonic-buildimage/pull/5866

**Comment by gechiang** [comment]
Validated with Master image .496 and it looks like this issue is fixed.
